diffrax/_custom_types.py

```diff
@@ -53,34 +52,41 @@
 BufferDenseInfos = dict[str, PyTree[eqxi.MaybeBuffer[Shaped[Array, "times ..."]]]]
 sentinel: Any = eqxi.doc_repr(object(), "sentinel")
 
-LevyArea: TypeAlias = Literal["", "space-time"]
 
+class AbstractLevyArea(eqx.Module):
+    dt: eqx.AbstractVar[PyTree]
+    W: eqx.AbstractVar[PyTree]
 
-class LevyVal(eqx.Module):
+
+class TimeLevyArea(AbstractLevyArea):
+    dt: PyTree
+    W: PyTree
+
+
+class SpaceTimeLevyArea(AbstractLevyArea):
     dt: PyTree
     W: PyTree
-    H: Optional[PyTree]
-    bar_H: Optional[PyTree]
-    K: Optional[PyTree]
-    bar_K: Optional[PyTree]
+    H: PyTree
+    bar_H: Optional[PyTree] = None
+    K: Optional[PyTree] = None
+    bar_K: Optional[PyTree] = None
 
 
-def levy_tree_transpose(tree_shape, levy_area: LevyArea, tree: PyTree):
+def levy_tree_transpose(tree_shape, levy_area: type[AbstractLevyArea], tree: PyTree):
     """Helper that takes a PyTree of LevyVals and transposes
     into a LevyVal of PyTrees.
 
     **Arguments:**
 
     - `tree_shape`: Corresponds to `outer_treedef` in `jax.tree_transpose`.
-    - `levy_area`: can be `""` or `"space-time"`, which indicates
-    which fields of the LevyVal will have values.
+    - `levy_area`: the type of `AbstractLevyReturn`
```

Comment:
"`AbstractLevyReturn`" isn't used as a name anywhere else.

Reply:
Given the decision to use `BrownianIncrement`, do you want to use the `AbstractLevyReturn` or `AbstractBrownianReturn` for this type?

Reply:
Let's go for `AbstractBrownianReturn`!